lib/server.js

```diff
@@ -62,13 +62,23 @@ const screenUsers = (ENV) => {
   // generic handling
   const allowedRegex = /\.[^.]{2,3}(\?[^?]+){0,1}$/
   router.use((req, res, next) => {
+    // Avoid double slashes by merging 2 or more into one, before continuing
+    // with the following rules. This mitigates a redirection vulnerability.
+    // Examples:
+    //   service.com//google.co.uk/ -> service.com/google.co.uk/
+    //   service.com/a//b/c///d/ -> service.com/a/b/c/d/
+    //
+    req.url = req.url.replace(/\/+/g, '/')
```

Comment:
It's a bit hard for me to see if the `req.url` variable contains the `https://`
 (double slash) bit in there, if so then the url would become `https:**/**service.com/google.com` (single slashed)

My suggestion is to use `/([^:])(\/\/+)/g, '$1/'`, a bit more complex to read but will serve you in either way, example:

```
var str1 = 'https://service.com//google.com'
var str2 = 'service.com//google.com';


var clean_url = str1.replace(/([^:])(\/\/+)/g, '$1/');
alert(clean_url);
> https://service.com/google.com

var clean_url = str2.replace(/([^:])(\/\/+)/g, '$1/');
alert(clean_url);
> service.com/google.com

```

Reply:
The req.url does not contains http[s]:// etc.
Only the path.

So it will contain: `//google.com/` or `//a//b/c///d` etc.

Reply:
I will add a bit more clarification in the comment. I get that `request.url` sounds like it contains the full URL, so it is not completely clear.